internal/sigstore/sign.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+package sigstore
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"errors"
+	"fmt"
+
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/theupdateframework/go-tuf/encrypted"
+)
+
+const (
+	cosignPrivateKeyPemType   = "ENCRYPTED COSIGN PRIVATE KEY"
+	sigstorePrivateKeyPemType = "ENCRYPTED SIGSTORE PRIVATE KEY"
+)
+
+// SignContent signs the content with the cosign encrypted private key and corresponding cosign password.
+func SignContent(password, encryptedPrivateKey, content []byte) ([]byte, error) {
```

Comment:
Is the cosign private key encrypted in our CI?
If not (or if we can save it unencrypted, which shouldnt make a difference because saving passphrase + encrypted data is basically the same as just saving the unencrypted data) we can remove the extra steps of parsing passphrase the passphrase and decrypting the key.
This should also remove some dependencies.

Reply:
> Is the cosign private key encrypted in our CI?

I think so. COSIGN_PASSWORD is a secret referenced in gh actions

Reply:
yes, both in the CI and keepass they are encrypted. There is no external dependency anymore.

Reply:
We now have a dependency on `github.com/theupdateframework/go-tuf/encrypted`, but since this was already an indirect dependency previously, there shouldnt be a difference really.

If the cosign key wasnt additionally encrypted, we could skip the decrypting part in this function.
But doesnt make a big difference imo